### PR TITLE
Some minor refactoring.

### PR DESCRIPTION
**Pull request by ZehMatt**
This PR renames a few variables to the best guessed name by its surroundings. I also managed to find some unused variables which lead to some unnecessary code which I also remove in this PR. I plan on making more of those smaller PRs getting the code more or less readable.

### CODE REVIEW

File: src/openrct2/paint/paint.c [hunk 1]
```diff
@@ -46,13 +46,13 @@ static uint32 _paintQuadrantFrontIndex;
 static paint_struct *_paintQuadrants[MAX_PAINT_QUADRANTS];
 void *g_currently_drawn_item;
 paint_entry * gEndOfPaintStructArray;
-sint16 gUnk9DE568;
-sint16 gUnk9DE56C;
+rct_xy16 gPaintSpritePosition;
+// sint16 gUnk9DE568;
```
Comment (Gymnasiast) [inline src/openrct2/paint/paint.c]: Probably better to just remove this, rather than commenting out.

Comment (ZehMatt) [reply src/openrct2/paint/paint.c]: Ops, that was not my intention.

**Comment by Gymnasiast** [comment]
Due to your changes, FadeSprites in Sprite.cpp is now unused. Clang fails because of this.

**Comment by ZehMatt** [comment]
Will do.

**Comment by duncanspumpkin** [comment]
There was a good reason why i didn't refactor out `9ABDA4` from the sprite interaction code. I think it was because it didn't need to be global. You will notice it was completely removed from the actual drawing code. I think this can still be worked on to remove its need completely. 

Your removal of `9ABDAE` brings up an issue. I expect that is saying that RLE sprites will not interact correctly at increased zoom levels.

File: src/openrct2/interface/viewport.c [hunk 2]
```diff
@@ -1236,21 +1221,17 @@ static bool sub_679074(rct_drawpixelinfo *dpi, sint32 imageId, sint16 x, sint16
...
-    uint32 ebx = gUnkEDF81C;
+    uint32 imageType = gCurrentImageType;
 
-    if (!(image->flags & 2)) {
-        return sub_679236_679662_679B0D_679FF1(ebx, image, offset);
+    if (!(image->flags & G1_FLAG_1)) {
+        return sub_679236_679662_679B0D_679FF1(imageType, image, offset);
     }
 
     // The code below is untested.
```
Comment (duncanspumpkin) [inline src/openrct2/interface/viewport.c]: Please add an assert to this code. I don't think this code is needed and can be removed but best run it with an assert for a while before removal. I think we have the same in the drawing code.

File: src/openrct2/interface/viewport.c [hunk 3]
```diff
@@ -1032,26 +1029,21 @@ static void store_interaction_info(paint_struct *ps)
 /**
  * rct2: 0x00679236, 0x00679662, 0x00679B0D, 0x00679FF1
  */
-static bool sub_679236_679662_679B0D_679FF1(uint32 ebx, rct_g1_element *image, uint8 *esi) {
+static bool sub_679236_679662_679B0D_679FF1(uint32 imageType, rct_g1_element *image, uint8 *index) {
```
Comment (duncanspumpkin) [inline src/openrct2/interface/viewport.c]: This should be renamed to make it clear that this is testing to see if a pixel can be interacted with.

Comment (ZehMatt) [reply src/openrct2/interface/viewport.c]: I wasn't exactly sure what its doing, any suggestions on the name?

File: src/openrct2/interface/viewport.c [hunk 4]
```diff
@@ -1236,21 +1221,19 @@ static bool sub_679074(rct_drawpixelinfo *dpi, sint32 imageId, sint16 x, sint16
...
 
     uint8 *offset = image->offset + (yStartPoint * image->width) + xStartPoint;
-    uint32 ebx = gUnkEDF81C;
+    uint32 imageType = gCurrentImageType;
 
-    if (!(image->flags & 2)) {
-        return sub_679236_679662_679B0D_679FF1(ebx, image, offset);
+    if (!(image->flags & G1_FLAG_1)) {
+        // Using assert here because it is probably unused, if not triggered remove this block.
+        assert(false);
```
Comment (duncanspumpkin) [inline src/openrct2/interface/viewport.c]: I think you miss understood me. This bit of code is perfectly fine its the block below this that starts with the comment `The code below is untested.` that is the dead code. I've also made a small mistake with the name when i discussed this on gitter. `sub_679236_679662_679B0D_679FF1` is the bmp `pixel_is_present` function and `sub_67933B_679788_679C4A_67A117` is the rle `pixel_is_present` function.

**Comment by duncanspumpkin** [comment]
Looks good 😁

**Comment by Gymnasiast** [comment]
So it's ready for merging, @duncanspumpkin ?

File: src/openrct2/paint/paint.h [hunk 5]
```diff
@@ -136,15 +136,13 @@ typedef struct support_height {
...
-extern sint16 gUnk9DE56C;
+extern rct_xy16 gPaintSpritePosition;
 extern paint_entry gPaintStructs[4000];
 #else
 #define gPaintStructs RCT2_ADDRESS(0x00EE788C, paint_entry)
 #define g_currently_drawn_item  RCT2_GLOBAL(0x009DE578, void*)
 #define gEndOfPaintStructArray  RCT2_GLOBAL(0x00EE7880, paint_entry *)
-#define gUnk9DE568              RCT2_GLOBAL(0x009DE568, sint16)
-#define gUnk9DE56C              RCT2_GLOBAL(0x009DE56C, sint16)
+#define gPaintSpritePosition    RCT2_GLOBAL(0x009DE568, rct_xy16)
```
Comment (IntelOrca) [inline src/openrct2/paint/paint.h]: This is incorrect, the original addresses are 4 bytes apart, not two - so a `rct_xy16` does not map on to them.

**Pull request merged**